backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs

```diff
@@ -106,17 +121,19 @@ public async Task<List<UserDto>> GetProjectUsersAsync(int projectId)
     public async Task<List<ProjectResponseDto>> GetAllUserProjectsAsync()
     {
         var currentUserId = _userIdGetter.GetCurrentUserId();
-        var userProjects = await _context.Projects
+        var userProjects = await _context.Users
+            .Where(user => user.Id == currentUserId)
+            .SelectMany(user => user.Projects)
             .Include(project => project.Tags)
-            .Where(x => x.CreatedBy == currentUserId)
             .ToListAsync();
 
+
         return _mapper.Map<List<ProjectResponseDto>>(userProjects)!;
     }
 
     private void ValidateProject(Project? entity)
     {
-        if (entity is null || entity.CreatedBy != _userIdGetter.GetCurrentUserId())
+        if (entity is null || entity.Users.All(user => user.Id != _userIdGetter.GetCurrentUserId()))
         {
             throw new EntityNotFoundException(nameof(Project));
         }
```

Comment:
also add check for current user as author of the project `(entity.Users.All(user => user.Id != _userIdGetter.GetCurrentUserId() && entity.CreatedBy != _userIdGetter.GetCurrentUserId())`